tests/integration/cmd_debug_test.go

```diff
@@ -0,0 +1,66 @@
+package integration
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/odo/tests/helper"
+)
+
+var _ = Describe("odo debug command tests", func() {
+	var project string
+	var context string
+
+	// Setup up state for each test spec
+	// create new project (not set as active) and new context directory for each test spec
+	// This is before every spec (It)
+	BeforeEach(func() {
+		SetDefaultEventuallyTimeout(10 * time.Minute)
+		SetDefaultConsistentlyDuration(30 * time.Second)
+		context = helper.CreateNewContext()
+		project = helper.CreateRandProject()
+		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
+	})
+
+	// Clean up after the test
+	// This is run after every Spec (It)
+	AfterEach(func() {
+		helper.DeleteProject(project)
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("odo debug on a nodejs component", func() {
+		It("should expect a ws connection when tried to connect on different debug port locally and remotely", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), context)
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", "--project", project, "--context", context)
+			helper.CmdShouldPass("odo", "config", "set", "--force", "DebugPort", "9292", "--context", context)
+			dbgPort := helper.GetConfigValueWithContext("DebugPort", context)
+			Expect(dbgPort).To(Equal("9292"))
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			go func() {
+				helper.CmdShouldRunWithTimeout(60*time.Second, "odo", "debug", "port-forward", "--local-port", "5050", "--context", context)
+			}()
+
+			// debug port
+			helper.HttpWaitForWithStatus("http://localhost:5050", "WebSockets request was expected", 12, 5, 400)
```

Comment:
Ideally it should listen through the port ```5050```, but here its returning status code of ```400```.

Reply:
This excepted because after timeout the session is being terminated. But this is not the proper way of validation IMO

Reply:
Yes. We are expecting HTTP 400 error code. NodeJS debugger uses WebSockets. Instead of creating a complicated WebSocket client, we just sent do a simple HTTP GET request. If there is something else on that port, it will fail, if the port is not open it will fail, if there is some mixup in the port numbers and the normal HTTP server is listening on that port, it will return 200 and test will fail.  If there is something that uses websocets it will return 400 and test is successful.

@girishramnani There should probably be a comment explaining why we are expecting 400, as it might be confusing for someone reading the code who is not familiar with this background.

Reply:
Thanks for the details

Reply:
Yes because the "expected ws request" is 400 response. It's the only way to validate